src/ResourceManager/Version2016_09_01/ProviderRegistrationFactory.cs

```diff
@@ -0,0 +1,100 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Management.Internal.Resources;
+using Microsoft.Azure.Management.Internal.Resources.Models;
+using Microsoft.WindowsAzure.Commands.Common;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Commands.ResourceManager.Common.Version2016_09_01
+{
+    public class ProviderRegistrationFactory : IProviderRegistrationFactory
+    {
+        class ProviderRegistrar : IProviderRegistrar
+        {
+            /// <summary>
+            /// Contains all providers we have attempted to register 
+            /// </summary>
+            private HashSet<string> _registeredProviders;
+
+            private Func<ResourceManagementClient> _createClient;
+
+            private Action<string> _writeDebug;
+
+            private ResourceManagementClient Client => _createClient();
```

Comment:
Are you intending to create a new client every time the Client property is accessed?

Reply:
It should create a client each time, as the authentication details may change between usages